scripts/vehicle_node.py

```diff
@@ -201,78 +203,115 @@ def update_position(self, event):
-        footprint = np.matmul(vehicle_T_joint1, footprint)
-
-        self.joint_pose.position.x = footprint[0,0]
-        self.joint_pose.position.y = footprint[1,0]
-        self.joint_pose.position.z = footprint[2]
-        self.joint_pose.orientation.x = q2[0]
-        self.joint_pose.orientation.y = q2[1]
-        self.joint_pose.orientation.z = q2[2]
-        self.joint_pose.orientation.w = q2[3]
+        #Calculate angle based on velocity data and time
+        angle = joint_data.velocity[0]*time
+        angle2 = joint_data.velocity[1]*time
+
+        self.joint1_pitch += angle
+        self.joint2_pitch += angle2
+
+        #Transformations from vehicle frame to Joint1 and Joint2
+    
+        #Static rotation about z-axis 
+        static_rot = tf.transformations.quaternion_from_euler(0.0, 0.0, 3.14159)
+        translation =  [0.0, 0.0, 0.0]
+        V_T_SRz = tf.transformations.quaternion_matrix(static_rot)
+        V_T_SRz[:3,3] = np.array(translation)
+
+
+        #Dynamic rotation of Joint 1
+        dynamic_rot = tf.transformations.quaternion_from_euler(0.0, self.joint1_pitch, 0.0)
+        translation =  [0.0, 0.0, 0.0]
+        SRz_T_J1 = tf.transformations.quaternion_matrix(dynamic_rot)
+        SRz_T_J1[:3,3] = np.array(translation)
+
+
+        #Translation based on length of Joint1 arm 
+        no_rot = tf.transformations.quaternion_from_euler(0.0, 0.0, 0.0)
+        translation =  [self.joint1_length, 0.0, 0.0]
+        J1_T_STx = tf.transformations.quaternion_matrix(no_rot)
+        J1_T_STx[:3,3] = np.array(translation)
+
+        #Dynamic rotation of Joint2
+        dynamic_rot2 = tf.transformations.quaternion_from_euler(0.0, self.joint2_pitch, 0.0)
+        translation =  [0.0, 0.0, 0.0]
+        STx_T_J2 = tf.transformations.quaternion_matrix(dynamic_rot2)
+        STx_T_J2[:3,3] = np.array(translation)
+
+        #matrix multiplication to form the homogenous matrices
+        V_T_J1 = np.matmul(V_T_SRz, SRz_T_J1)
+        V_T_STx = np.matmul(V_T_J1, J1_T_STx)
+        V_T_J2 = np.matmul(V_T_STx, STx_T_J2)
+        rot_J1 = tf.transformations.quaternion_from_matrix(V_T_J1)
+        rot_J2 = tf.transformations.quaternion_from_matrix(V_T_J2)
+
+        #Assign pose information for publishing
+        self.joint1_pose.position.x = V_T_J1[0][3]
+        self.joint1_pose.position.y = V_T_J1[1][3]
+        self.joint1_pose.position.z = V_T_J1[2][3]
+        self.joint1_pose.orientation.x = rot_J1[0]
+        self.joint1_pose.orientation.y = rot_J1[1]
+        self.joint1_pose.orientation.z = rot_J1[2]
+        self.joint1_pose.orientation.w = rot_J1[3]
+
+        self.joint2_pose.position.x = V_T_J2[0][3]
+        self.joint2_pose.position.y = V_T_J2[1][3]
+        self.joint2_pose.position.z = V_T_J2[2][3]
+        self.joint2_pose.orientation.x = rot_J2[0]
+        self.joint2_pose.orientation.y = rot_J2[1]
+        self.joint2_pose.orientation.z = rot_J2[2]
+        self.joint2_pose.orientation.w = rot_J2[3]
```

Comment:
This looks super repetitive.  Take a look at `tf_conversions` and see if there's any useful helper functions.
http://wiki.ros.org/tf_conversions
https://github.com/ros/geometry/blob/melodic-devel/tf_conversions/src/tf_conversions/posemath.py